### PR TITLE
Gracefully handle invalid sensor types in REST requests

### PR DESCRIPTION
**Pull request by algairim**
Signed-off-by: Mykola Mandra <mykola.mandra@cloudsoft.io>

### CODE REVIEW

File: rest/rest-resources/src/main/java/org/apache/brooklyn/rest/resources/ApplicationResource.java [hunk 1]
```diff
@@ -710,13 +711,7 @@ private void checkLocationsAreValid(ApplicationSpec applicationSpec) {
             return result;
         }
         for (Entity e: descs) {
-            Object v = null;
-            try {
-                v = e.getAttribute((AttributeSensor<?>)s);
-            } catch (Exception exc) {
-                Exceptions.propagateIfFatal(exc);
-                log.warn("Error retrieving sensor "+s+" for "+e+" (ignoring): "+exc);
-            }
```
Comment (algairim) [inline rest/rest-resources/src/main/java/org/apache/brooklyn/rest/resources/ApplicationResource.java:719]: This code has been moved into `EntityAttributesUtils` and re-used in all explicit attempts to retrieve a sensor in `rest-resources` project, see next line.

File: rest/rest-resources/src/main/java/org/apache/brooklyn/rest/util/EntityAttributesUtils.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ */
+package org.apache.brooklyn.rest.util;
+
+import org.apache.brooklyn.api.entity.Entity;
+import org.apache.brooklyn.api.sensor.AttributeSensor;
+import org.apache.brooklyn.util.exceptions.Exceptions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EntityAttributesUtils {
```
Comment (jcabrerizo) [inline rest/rest-resources/src/main/java/org/apache/brooklyn/rest/util/EntityAttributesUtils.java:27]: Logic is good, but in my opinion this method should be part of `BrooklynRestResourceUtils` 
What do you think?

**Review by jcabrerizo — CHANGES_REQUESTED** [review]
Looks good @algairim, just a couple of minor fix

File: rest/rest-resources/src/main/java/org/apache/brooklyn/rest/util/EntityAttributesUtils.java (hunk 2 above)
Comment (algairim) [reply rest/rest-resources/src/main/java/org/apache/brooklyn/rest/util/EntityAttributesUtils.java:27]: This is consistent with existing standalone utils and has nothing to do with `BrooklynRestResourceUtils`, which is used in in transformers to pull items from the management context.

File: rest/rest-resources/src/main/java/org/apache/brooklyn/rest/resources/ScriptResource.java [hunk 3]
```diff
@@ -26,6 +26,7 @@
 import org.apache.brooklyn.core.mgmt.entitlement.Entitlements;
 import org.apache.brooklyn.rest.api.ScriptApi;
 import org.apache.brooklyn.rest.domain.ScriptExecutionSummary;
+import org.apache.brooklyn.rest.util.EntityAttributesUtils;
```
[the remark this answers is not in the document]
Comment (algairim) [reply rest/rest-resources/src/main/java/org/apache/brooklyn/rest/resources/ScriptResource.java]: e6305e9a8df01b2ae5252609d849f3ededc020ff

File: rest/rest-resources/src/main/java/org/apache/brooklyn/rest/resources/SensorResource.java [hunk 4]
```diff
@@ -26,14 +26,17 @@
 import org.apache.brooklyn.api.entity.Entity;
 import org.apache.brooklyn.api.sensor.AttributeSensor;
 import org.apache.brooklyn.api.sensor.Sensor;
+import org.apache.brooklyn.core.entity.Attributes;
 import org.apache.brooklyn.core.entity.EntityInternal;
+import org.apache.brooklyn.core.entity.lifecycle.Lifecycle;
```
[the remark this answers is not in the document]
Comment (algairim) [reply rest/rest-resources/src/main/java/org/apache/brooklyn/rest/resources/SensorResource.java]: removed - e6305e9a8df01b2ae5252609d849f3ededc020ff

**Comment by jcabrerizo** [comment]
Thank for this and your comments @algairim

**Pull request merged**